src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs

```diff
@@ -8,18 +8,26 @@
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Text.Unicode;
+
+#if SYSTEM_PRIVATE_CORELIB
 using Internal.Runtime.CompilerServices;
+#endif
 
 #pragma warning disable 0809  //warning CS0809: Obsolete member 'Utf8Span.Equals(object)' overrides non-obsolete member 'object.Equals(object)'
 
 #pragma warning disable SA1121 // explicitly using type aliases instead of built-in types
+#if SYSTEM_PRIVATE_CORELIB
 #if TARGET_64BIT
 using nint = System.Int64;
 using nuint = System.UInt64;
 #else
 using nint = System.Int32;
 using nuint = System.UInt32;
 #endif
+#else
+using nint = System.Int64;
```

Comment:
These same defines exist in a bunch of the _Utf8Utility.\*.cs_ and _Utf8.\*.cs_ files. Are we sure the behavior will be correct for all bitnesses when running on .NET Framework?

Reply:
The current plan is to use `long`/`ulong` for now when we are outside of CoreLib. This could result in worse perf on 32-bit processes than if we used `int`/`uint`. But if we used `int`, it could result in worse perf for 64-bit processes. Picking `long` for now, since we need to pick one, and we know it will always be big enough.

I've logged https://github.com/dotnet/runtime/issues/33575 to track using the new native-int feature when it is available.